### PR TITLE
Add dialog to userProfile

### LINKED ISSUE #81: Make a dialog list for followers in userProfile

### PR DESCRIPTION
**Pull request by alshakerM**
Added dialog to userProfile.

Fixes #81

### CODE REVIEW

**Review by alshakero — CHANGES_REQUESTED** [review]
Two tiny issues. Other than that, it looks perfect.

File: components/pages/UserProfile/UserProfile.module.css [hunk 1]
```diff
@@ -189,3 +195,60 @@
...
+
+.dialogTitle {
+  display: flex;
+  align-items: center;
+  border-bottom: 1px solid #dbdbdb;
+  height: 43px;
+  font-size: 16px;
+  font-weight: 550;
+  color: #262626;
+  border-radius: 12px;
```
[the remark this answers is not in the document]
Comment (alshakerM) [reply components/pages/UserProfile/UserProfile.module.css:226]: Yea, ran into a few MUI issues, forgot to remove it.

**Comment by alshakerM** [comment]
Removed console, log.
Removed border radius.
Added cursor pointer to clear button.
Changed action type name.

File: components/pages/UserProfile/UserProfile.js [hunk 2]
```diff
@@ -1,15 +1,80 @@
...
+            ></line>
+          </svg>
+        </button>
+      </DialogTitle>
+      <DialogContent className={styles.dialogContent}>
+        {users?.map((data) => (
+          <div className={styles.followersSection}>
+            <Avatar user={data} alt="profile pic" size="36" />
+            <div className={styles.followerInfo}>
+              <p className={styles.followersUserName}>{data.user_name}</p>
```
Comment (alshakero) [inline components/pages/UserProfile/UserProfile.js:68]: This should be a link to the user's profile.

**Review by alshakero — APPROVED** [review]
I really like this PR! One tiny issue.

**Comment by alshakerM** [comment]
Fixed a bug with the store.
Added a link to Avatar to profile page or stories

File: components/Avatar/Avatar.js [hunk 3]
```diff
@@ -10,6 +10,7 @@ export function Avatar({
   animate,
   colorRing,
   isUnread = true,
+  link = `/stories/${user.user_id}`,
```
Comment (alshakero) [inline components/Avatar/Avatar.js]: What if the user has no stories?

File: components/Avatar/Avatar.js [hunk 4]
```diff
@@ -32,7 +33,7 @@ export function Avatar({
   }
 
   return (
-    <Link href={`/stories/${user.user_id}`}>
+    <Link href={link}>
```
Comment (alshakero) [inline components/Avatar/Avatar.js]: We discussed this one.

File: stores/profileStore/reducer.js [hunk 5]
```diff
@@ -1,20 +1,25 @@
...
+        if (!draft.fullProfilesData[username]) {
+          draft.fullProfilesData[username] = {};
+        }
+        draft.fullProfilesData[username][postType] = action.fullProfileData;
+      });
     }
-    case 'RECEIVE_ALL_PROFILES': {
-      return { ...state, profileData: action.profileData };
+    case 'RECEIVE_PROFILES': {
+      debugger
```
Comment (alshakero) [inline stores/profileStore/reducer.js:21]: --

**Review by alshakero — CHANGES_REQUESTED** [review]
Tiny issue.

**Review by alshakero — APPROVED** [review]
LGTM.

File: components/Avatar/Avatar.js (hunk 3 above)
Comment (alshakerM) [reply components/Avatar/Avatar.js]: This code is from an old commit.
This issue was fixed in a newer commit.
https://github.com/alshakerM/my-instagram/pull/83/commits/767952d3d093835b725e014942742c609831abc0

File: components/Avatar/Avatar.js (hunk 4 above)
Comment (alshakerM) [reply components/Avatar/Avatar.js]: This code is from an old commit.
This issue was fixed in a newer commit.
https://github.com/alshakerM/my-instagram/pull/83/commits/767952d3d093835b725e014942742c609831abc0

File: stores/profileStore/reducer.js (hunk 5 above)
Comment (alshakerM) [reply stores/profileStore/reducer.js:21]: Addressed in 1 to 1.

**Review by alshakero — APPROVED** [review]
Perfect now.

**Pull request merged**